### PR TITLE
DEB-9: Patient immunization

### PR DESCRIPTION
**Pull request by bradsawadye**
DEB-9

### CODE REVIEW

**Comment by MattyJ007** [comment]
@bradsawadye This is not an acceptable Commit message.
See the examples in our ways of [working document](https://docs.google.com/document/d/1fiFr1q4tk4sh6K6Sd7g0hjB8shGMfmY-ZNE3c8qG3kg/edit)

File: docker/importer/volume/openhim-import.json [hunk 1]
```diff
@@ -1,52 +1,61 @@
...
+      "autoRetryPeriodMinutes": 60,
+      "routes": [
         {
           "type": "http",
           "status": "enabled",
           "forwardAuthHeader": false,
-          "name": "Generic Mapping Mediator",
+          "name": "Hapi Fhir",
+          "secured": false,
           "host": "localhost",
```
Comment (de-laz) [inline docker/importer/volume/openhim-import.json]: @bradsawadye Looks good man. Is there any reason why we are not using the docker compose service name for the FHIR server here? I believe it's `hapi-fhir`, the service name

**Review by de-laz — CHANGES_REQUESTED** [review]

File: docker/importer/volume/openhim-import.json (hunk 1 above)
Comment (bradsawadye) [reply docker/importer/volume/openhim-import.json]: @de-laz thanks for the feedback. The issue has been addressed in https://github.com/jembi/covid19-immunization-tracking-package/pull/2/commits/a4db5873c692af2983bd13b3047fbf1a2b7cede2

Comment (bradsawadye) [reply docker/importer/volume/openhim-import.json]: @de-laz thanks for the feedback. The issue has been addressed in https://github.com/jembi/covid19-immunization-tracking-package/pull/2/commits/a4db5873c692af2983bd13b3047fbf1a2b7cede2

File: features/immunization/utils.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+const {
+  sendRequest,
+  verifyResourceExistsAndCleanup,
+  verifyOpenhimIsRunning
+} = require('../utils')
+
+let authorizationError
+
+const immunization = JSON.parse(
+  fs.readFileSync(`${__dirname}/payloads/immunization.json`, 'utf8')
```
Comment (MattyJ007) [inline features/immunization/utils.js]: @bradsawadye This directory doesn't exist

File: package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "covid19-immunization-tracking-package",
+  "version": "0.0.1",
+  "main": "index.js",
+  "scripts": {
+    "lint": "eslint ./features/*/*.js",
+    "lint:fix": "eslint ./features/*/*.js --fix",
```
Comment (MattyJ007) [inline package.json:7]: These two commands don't work

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
@bradsawadye There are broken commands in these scripts

File: package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "test": "env-cmd -f ./docker/.env.local cucumber-js -f progress-bar ./features/",
+    "test:remote": "env-cmd -f ./ansible/.env.remote cucumber-js -f progress-bar ./features/"
```
Comment (MattyJ007) [inline package.json]: @bradsawadye This is not an ansible project

File: docker/importer/volume/openhim-import.json [hunk 5]
```diff
@@ -1,52 +1,118 @@
...
-      "urn": "covid19-immunization-tracking-mediator",
-      "version": "2.2.2",
-      "name": "Mapping Mediator",
-      "description": "Generic OpenHIM Mapping Mediator",
-      "endpoints": [
+      "methods": [
+        "GET",
+        "POST",
+        "DELETE",
+        "PUT"
```
Comment (MattyJ007) [inline docker/importer/volume/openhim-import.json:65]: @bradsawadye This is a search channel - only GET

Comment (MattyJ007) [inline docker/importer/volume/openhim-import.json]: @bradsawadye This is a search channel only - only GET

**Review by MattyJ007 — CHANGES_REQUESTED** [review]
@bradsawadye 
Looking good for the patient search channel just one small change 👍🏼

File: README.md [hunk 6]
```diff
@@ -64,3 +71,8 @@ TODO: Channel config.
   "birthDate": "1981-05-21"
 }
 ```
+
+You should receive the created form of the Patient resource in HAPI FHIR - notice the Resource now has an `id` field.
+
+To **update** a patient, send through the payload above to `/patient-registration/{FHIR_ID}` but add the field `id` to the object root.
+Also substitute in the `id` into the Request path in the `FHIR_ID` placeholder position.
```
Comment (MattyJ007) [inline README.md:78]: @bradsawadye Could you please add some examples of the Patient search feature here as well as the immunization

**Review by MattyJ007 — APPROVED** [review]
@bradsawadye I'm approving this PR so that my `requested changes` don't block the PR whilst I'm on leave.
But there are still a few changes that are needed -
@de-laz could you please test my change requests when they're implemented

**Review by de-laz — APPROVED** [review]

**Comment by de-laz** [comment]
@bradsawadye As discussed, I will be merging this branch and the remaining work will be done on a new branch.

**Pull request merged**